### PR TITLE
Add JWT Support Task

### PR DESCRIPTION
**Pull request by RomanMenzheres**
Added JWT support, open api available for all now, created 'OpenApiConfig' to add to swagger JWT support.

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -2,24 +2,32 @@
...
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
 @EnableMethodSecurity
 @Configuration
 public class SecurityConfig {
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java:23]: Empty line after declaration of class, for what?

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 2]
```diff
@@ -34,13 +42,27 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(
                         auth -> auth
-                                .requestMatchers("/api/auth/**", "/error")
+                                .requestMatchers("/api/auth/**", "/swagger-ui/index.html", "/error")
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java:45]:
```suggestion
                                .requestMatchers("/api/auth/**", "/swagger-ui/**", "/error")
```

Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java:45]: What is /error?

File: src/main/java/com/example/bookstore/dto/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotEmpty;
+
+public class UserLoginRequestDto {
+    @NotEmpty
+    @Email
+    private String email;
+    @NotEmpty
+    @Min(8)
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/dto/UserLoginRequestDto.java:12]: Could you validate w/o NotEmpty anno?

File: src/main/java/com/example/bookstore/security/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+@Service
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public AuthenticationService(JwtUtil jwtUtil, AuthenticationManager authenticationManager) {
+        this.jwtUtil = jwtUtil;
+        this.authenticationManager = authenticationManager;
+    }
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/security/AuthenticationService.java:18]: You can use lombok @requiredArgsConstructor instead this constructor

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java:18]:
```suggestion
@Component
@RequiredArgsConstructor
public class JwtAuthenticationFilter extends OncePerRequestFilter {
```

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    private final JwtUtil jwtUtil;
+    private final UserDetailsService userDetailsService;
+
+    public JwtAuthenticationFilter(JwtUtil jwtUtil,
+                                   UserDetailsService userDetailsService) {
+        this.jwtUtil = jwtUtil;
+        this.userDetailsService = userDetailsService;
+    }
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java:26]:
```suggestion
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**